diff -ur sqlite-3.6.4.org/Makefile.in sqlite-3.6.4/Makefile.in
--- sqlite-3.6.4.org/Makefile.in	2008-10-14 20:07:37.000000000 +0200
+++ sqlite-3.6.4/Makefile.in	2008-10-21 16:57:17.000000000 +0200
@@ -15,7 +15,10 @@
 # The toplevel directory of the source tree.  This is the directory
 # that contains this "Makefile.in" and the "configure.in" script.
 #
-TOP = @srcdir@
+TOP = $(srcdir)
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+top_builddir = .
 
 # C Compiler and options for use in building executables that
 # will run on the platform that is doing the build.
@@ -139,6 +146,6 @@
 bindir = @bindir@
 includedir = @includedir@
 INSTALL = @INSTALL@
-LIBTOOL = ./libtool
+LIBTOOL = @LIBTOOL@
 ALLOWRELEASE = @ALLOWRELEASE@
 
