diff -ur sqlite-3.5.9.org2/Makefile.in sqlite-3.5.9/Makefile.in
--- sqlite-3.5.9.org2/Makefile.in	2008-07-02 10:55:12.000000000 +0200
+++ sqlite-3.5.9/Makefile.in	2008-07-02 11:18:13.000000000 +0200
@@ -36,6 +36,10 @@
 # 
 TCC += -D_HAVE_SQLITE_CONFIG_H
 
+# OE overrides
+#
+TARGET_LFLAGS = @TARGET_LFLAGS@
+
 # Define -DNDEBUG to compile without debugging (i.e., for production usage)
 # Omitting the define will cause extra debugging code to be inserted and
 # includes extra comments when "EXPLAIN stmt" is used.
@@ -415,17 +419,17 @@
           | $(NAWK) '{print $$5,$$6}' >last_change
 
 libsqlite3.la:	$(LIBOBJ)
-	$(LTLINK) -o $@ $(LIBOBJ) $(TLIBS) \
+	$(LTLINK) -o $@ $(LIBOBJ) $(TARGET_LFLAGS) $(TLIBS) \
 		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -o $@ tclsqlite.lo \
-		$(LIBOBJ) @TCL_STUB_LIB_SPEC@ $(TLIBS) \
+		$(LIBOBJ) @TCL_STUB_LIB_SPEC@ $(TARGET_LFLAGS) $(TLIBS) \
                 -rpath "$(libdir)/sqlite" \
 		-version-info "8:6:8"
 
 sqlite3$(TEXE):	$(TOP)/src/shell.c libsqlite3.la sqlite3.h
-	$(LTLINK) $(READLINE_FLAGS) \
+	$(LTLINK) $(TARGET_LFLAGS) $(READLINE_FLAGS) \
 		-o $@ $(TOP)/src/shell.c libsqlite3.la \
 		$(LIBREADLINE) $(TLIBS) -rpath "$(libdir)"
 
@@ -666,13 +670,13 @@
 
 tclsqlite3$(TEXE):	tclsqlite-shell.lo libsqlite3.la
 	$(LTLINK) -o $@ tclsqlite-shell.lo \
-		 libsqlite3.la $(LIBTCL)
+		 libsqlite3.la $(TARGET_LFLAGS) $(LIBTCL)
 
 testfixture$(TEXE):	$(TESTFIXTURE_SRC)
 	$(LTLINK) -DTCLSH=1 -DSQLITE_TEST=1 -DSQLITE_NO_SYNC=1\
 		-DSQLITE_CRASH_TEST=1 \
                 -DSQLITE_SERVER=1 -DSQLITE_PRIVATE="" -DSQLITE_CORE $(TEMP_STORE) \
-		-o $@ $(TESTFIXTURE_SRC) $(LIBTCL) $(TLIBS)
+		-o $@ $(TESTFIXTURE_SRC) $(TARGET_LFLAGS) $(LIBTCL) $(TLIBS)
 
 
 fulltest:	testfixture$(TEXE) sqlite3$(TEXE)
@@ -681,7 +685,7 @@
 test:	testfixture$(TEXE) sqlite3$(TEXE)
 	./testfixture$(TEXE) $(TOP)/test/quick.test
 
-sqlite3_analyzer$(TEXE):	$(TESTFIXTURE_SRC) $(TOP)/tool/spaceanal.tcl
+sqlite3_analyzer$(TEXE):	$(TARGET_LFLAGS) $(TESTFIXTURE_SRC) $(TOP)/tool/spaceanal.tcl
 	sed \
 	  -e '/^#/d' \
 	  -e 's,\\,\\\\,g' \
@@ -691,7 +695,7 @@
 	  $(TOP)/tool/spaceanal.tcl >spaceanal_tcl.h
 	$(LTLINK) -DTCLSH=2 -DSQLITE_TEST=1 -DSQLITE_CRASH_TEST=1 \
 		-DSQLITE_SERVER=1 -DSQLITE_PRIVATE="" -DSQLITE_CORE \
-		$(TEMP_STORE) -o $@ $(TESTFIXTURE_SRC) $(LIBTCL)
+		$(TEMP_STORE) -o $@ $(TESTFIXTURE_SRC) $(TARGET_LFLAGS) $(LIBTCL)
 
 
 install:	sqlite3$(BEXE) libsqlite3.la sqlite3.h ${HAVE_TCL:1=tcl_install}
