diff -Naur freetype-2.0.9.orig/builds/unix/freetype-config.in freetype-2.0.9/builds/unix/freetype-config.in
--- freetype-2.0.9.orig/builds/unix/freetype-config.in	Sat Feb  2 16:05:38 2002
+++ freetype-2.0.9/builds/unix/freetype-config.in	Thu Sep 19 01:27:45 2002
@@ -80,11 +80,7 @@
 
 if test "$echo_cflags" = "yes" ; then
  cflags="-I@includedir@/freetype2"
- if test "@includedir@" != "/usr/include" ; then
-  echo -I@includedir@ $cflags
- else
   echo $cflags
- fi
 fi
 
 if test "$echo_libs" = "yes" ; then
