diff -Naur ltrace-0.3.35.org/Makefile.in ltrace-0.3.35/Makefile.in
--- ltrace-0.3.35.org/Makefile.in	2003-03-30 16:38:05.000000000 +0200
+++ ltrace-0.3.35/Makefile.in	2005-02-01 18:31:46.000000000 +0100
@@ -49,7 +49,7 @@
 install:	ltrace
 		@INSTALL@ -d $(DESTDIR)$(bindir) $(DESTDIR)$(docdir) $(DESTDIR)$(mandir)/man1
 		@INSTALL@ -d $(DESTDIR)$(sysconfdir)
-		@INSTALL@ -s ltrace $(DESTDIR)$(bindir)
+		@INSTALL@ -m 755 ltrace $(DESTDIR)$(bindir)
 		@INSTALL@ -m 644 etc/ltrace.conf $(DESTDIR)$(sysconfdir)
 		@INSTALL@ -m 644 COPYING README TODO BUGS ChangeLog $(DESTDIR)$(docdir)
 		@INSTALL@ -m 644 ltrace.1 $(DESTDIR)$(mandir)/man1
diff -Naur ltrace-0.3.35.org/options.c ltrace-0.3.35/options.c
--- ltrace-0.3.35.org/options.c	2004-06-14 17:17:52.000000000 +0200
+++ ltrace-0.3.35/options.c	2005-02-01 18:36:07.000000000 +0100
@@ -12,6 +12,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <limits.h>
+#include <sys/param.h>
 
 #if HAVE_GETOPT_H
 #include <getopt.h>
diff -Naur ltrace-0.3.35.org/sysdeps/linux-gnu/Makefile ltrace-0.3.35/sysdeps/linux-gnu/Makefile
--- ltrace-0.3.35.org/sysdeps/linux-gnu/Makefile	2002-03-03 02:18:47.000000000 +0100
+++ ltrace-0.3.35/sysdeps/linux-gnu/Makefile	2005-02-01 18:31:46.000000000 +0100
@@ -1,4 +1,4 @@
-ARCH	:=	$(shell uname -m | sed -e s/i.86/i386/ -e s/sun4u/sparc64/ -e s/arm.*/arm/ -e s/sa110/arm/)
+ARCH	:=	ppc
 
 CFLAGS	+=	-I$(TOPDIR)/sysdeps/linux-gnu/$(ARCH)
 
