diff -Naur reiserfsprogs-3.6.14/include/credits.h bla/include/credits.h
--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ reiserfsprogs-3.6.14/include/credits.h	2004-05-21 09:22:37.000000000 +0200
@@ -0,0 +1,93 @@
+/*
+   Copyright 1996-2004 by Hans Reiser, licensing governed by 
+   reiserfsprogs/README.
+*/
+
+char *credits[] = {
+	"Hans Reiser was the project initiator,  source of all funding for the first 5.5\n"	\
+	"years. He is the architect and official maintainer.\n",
+
+	"Vladimir Saveliev started as the most junior programmer on the team, and became\n"	\
+	"the lead programmer.  He is now an experienced highly productive programmer. He\n"	\
+	"wrote the extent  handling code for Reiser4,  plus parts of  the balancing code \n"	\
+	"and file write and file read.\n",
+
+	"Alexander Zarochentcev  (zam)  wrote the high low priority locking code, online\n"	\
+	"resizer for V3 and V4, online repacker for V4, block allocation code, and major\n"	\
+	"parts of  the flush code,  and maintains the transaction manager code.  We give\n"	\
+	"him the stuff  that we know will be hard to debug,  or needs to be very cleanly\n"	\
+	"structured.\n",
+
+	"Nikita Danilov  wrote  most of the core  balancing code, plugin infrastructure,\n"	\
+	"and directory code. He steadily worked long hours, and is the reason so much of\n"	\
+	"the Reiser4 plugin infrastructure is well abstracted in its details.  The carry \n"	\
+	"function, and the use of non-recursive balancing, are his idea.\n",
+
+	"Vladimir Demidov wrote the parser for sys_reiser4(), the V3 alpha port, part of\n"	\
+	"the V3  journal  relocation code,  and helped  Hans keep  the business  side of\n"	\
+	"things running.\n",
+
+	"Chris Mason wrote the journaling code for V3,  which was enormously more useful\n"	\
+	"to users than just waiting until  we could create a wandering log filesystem as\n"	\
+	"Hans would have unwisely done without him.\n"	\
+	"Jeff Mahoney optimized the bitmap  scanning code for V3,  and performed the big\n"	\
+	"endian cleanups. \n",
+
+	"Elena Gryaznova performed testing and benchmarking.\n",
+
+	"Oleg Drokin was the debugger for  V3 during most of the time that  V4 was under\n"	\
+	"development,  and was quite  skilled and fast at it.  He wrote  the large write\n"	\
+	"optimization of V3.\n",
+
+	"Edward Shushkin wrote the encryption and compression  file plugins,  and the V3\n"	\
+	"journal relocation code.\n",
+
+	"Alexander  Lyamin  keeps our hardware  running,  and was very  generous  to our\n"	\
+	"project in many little ways.\n",
+
+	"Vitaly Fertman wrote  fsck for V3 and  maintains the reiserfsprogs package now.\n"	\
+	"He wrote librepair,  userspace plugins repair code, fsck for V4,  and worked on\n"	\
+	"developing libreiser4 and userspace plugins with Umka.\n",
+
+	"Yury Umanets  (aka Umka)  developed  libreiser4,  userspace  plugins,  and  all\n"	\
+	"userspace tools (reiser4progs) except of fsck.\n",
+
+	"Joshua Macdonald wrote the first draft of the transaction manager. Yuri Rupasov\n"	\
+	"did testing  and benchmarking,  plus he invented the r5 hash  (also used by the\n"	\
+	"dcache  code).  Yura  Rupasov,  Anatoly Pinchuk,  Igor Krasheninnikov,  Grigory\n"	\
+	"Zaigralin,  Mikhail  Gilula,   Igor  Zagorovsky,  Roman  Pozlevich,  Konstantin\n"	\
+	"Shvachko, and Joshua MacDonald are former contributors to the project.\n",
+
+	"Jeremy  Fitzhardinge  wrote  the  teahash.c  code  for  V3.  Colin  Plumb  also\n"	\
+	"contributed to that.\n",
+
+	"The  Defense  Advanced  Research  Projects Agency (DARPA, www.darpa.mil) is the\n"	\
+	"primary sponsor of Reiser4.  DARPA  does  not  endorse  this project; it merely \n"	\
+	"sponsors it.\n",
+
+	"Continuing core development of ReiserFS is  mostly paid for by Hans Reiser from\n"	\
+	"money made selling licenses  in addition to the GPL to companies who don't want\n"	\
+	"it known that they use ReiserFS  as a foundation for their proprietary product.\n"	\
+	"And my lawyer asked 'People pay you money for this?'. Yup. Life is good. If you\n"	\
+	"buy ReiserFS, you can focus on your value add rather than reinventing an entire\n"	\
+	"FS.\n",
+
+	"BigStorage  (www.bigstorage.com)  contributes to our general fund  every month,\n"	\
+	"and has done so for quite a long time.\n",
+
+	"SuSE (www.suse.com)  pays for continuing work on journaling for version 3, paid\n"	\
+	"for much of the previous version 3 work, and is paying for Chris and Jeff to do\n"	\
+	"V3 maintenance. Reiserfs integration in their distro is consistently solid, and\n"	\
+	"they were key to our becoming widely used.\n",
+
+	"Lycos (www.lycos.com) has a support contract with us that consistently comes in\n"	\
+	"just when  we would otherwise miss payroll,  and  that they keep doubling every \n"	\
+	"year. Much thanks to them.\n",
+
+	"Many persons came to www.namesys.com/support.html,  and got a question answered\n"	\
+	"for $25, or just gave us a small donation there.\n",
+
+	"Thanks to all  of those sponsors,  including  the secret ones.  Without you, we\n"	\
+	"would each still have that day job.\n"
+};
+#define CREDITS_COUNT 20
