--- sqlite-3.3.16/configure.ac.orig	2007-04-17 17:11:09.000000000 +0200
+++ sqlite-3.5.1/configure.ac	2007-04-17 17:13:52.000000000 +0200
@@ -336,6 +336,14 @@
 AC_SUBST(OS_OS2)
 AC_SUBST(TARGET_EXEEXT)
 
+if test "$config_TARGET_LFLAGS" != ""; then
+  TARGET_LFLAGS=$config_TARGET_LFLAGS
+  else
+    TARGET_LFLAGS=$BUILD_LFLAGS
+  fi
+AC_MSG_RESULT($TARGET_LFLAGS)
+AC_SUBST(TARGET_LFLAGS)
+
 ##########
 # Figure out all the parameters needed to compile against Tcl.
 #
@@ -482,64 +490,9 @@
 # Figure out what C libraries are required to compile programs
 # that use "readline()" library.
 #
-TARGET_READLINE_LIBS=""
+TARGET_READLINE_LIBS="-lreadline -lncurses"
 TARGET_READLINE_INC=""
-TARGET_HAVE_READLINE=0
-AC_ARG_ENABLE([readline],
-	[AC_HELP_STRING([--disable-readline],[disable readline support [default=detect]])],
-	[with_readline=$enableval],
-	[with_readline=auto])
-
-if test x"$with_readline" != xno; then
-	found="yes"
-
-	AC_ARG_WITH([readline-lib],
-		[AC_HELP_STRING([--with-readline-lib],[specify readline library])],
-		[with_readline_lib=$withval],
-		[with_readline_lib="auto"])
-	if test "x$with_readline_lib" = xauto; then
-		save_LIBS="$LIBS"
-		LIBS=""
-		AC_SEARCH_LIBS(tgetent, [readline ncurses curses termcap], [term_LIBS="$LIBS"], [term_LIBS=""])
-		AC_CHECK_LIB([readline], [readline], [TARGET_READLINE_LIBS="-lreadline"], [found="no"])
-		TARGET_READLINE_LIBS="$TARGET_READLINE_LIBS $term_LIBS"
-		LIBS="$save_LIBS"
-	else
-		TARGET_READLINE_LIBS="$with_readline_lib"
-	fi
-
-	AC_ARG_WITH([readline-inc],
-		[AC_HELP_STRING([--with-readline-inc],[specify readline include paths])],
-		[with_readline_inc=$withval],
-		[with_readline_inc="auto"])
-	if test "x$with_readline_inc" = xauto; then
-		AC_CHECK_HEADER(readline.h, [found="yes"], [
-			found="no"
-			if test "$cross_compiling" != yes; then
-				for dir in /usr /usr/local /usr/local/readline /usr/contrib /mingw; do
-					for subdir in include include/readline; do
-						AC_CHECK_FILE($dir/$subdir/readline.h, found=yes)
-						if test "$found" = "yes"; then
-							TARGET_READLINE_INC="-I$dir/$subdir"
-							break
-						fi
-					done
-					test "$found" = "yes" && break
-				done
-			fi
-		])
-	else
-		TARGET_READLINE_INC="$with_readline_inc"
-	fi
-
-	if test x"$found" = xno; then
-		TARGET_READLINE_LIBS=""
-		TARGET_READLINE_INC=""
-		TARGET_HAVE_READLINE=0
-	else
-		TARGET_HAVE_READLINE=1
-	fi
-fi
+TARGET_HAVE_READLINE=1
 
 AC_SUBST(TARGET_READLINE_LIBS)
 AC_SUBST(TARGET_READLINE_INC)
