diff -ur busybox-1.01/include/usage.h busybox-1.01.new/include/usage.h
--- busybox-1.01/include/usage.h	2008-03-06 16:38:22.000000000 +1300
+++ busybox-1.01.new/include/usage.h	2008-03-06 16:32:08.000000000 +1300
@@ -2829,7 +2829,8 @@
 
 #define wget_trivial_usage \
 	"[-c|--continue] [-q|--quiet] [-O|--output-document file]\n" \
-	"\t\t[--header 'header: value'] [-Y|--proxy on/off] [-P DIR] url"
+	"\t\t[--header 'header: value'] [-Y|--proxy on/off] [-P DIR]\n" \
+	"\t\t[-U|--user-agent agent] url"
 #define wget_full_usage \
 	"wget retrieves files via HTTP or FTP\n\n" \
 	"Options:\n" \
@@ -2837,6 +2838,7 @@
 	"\t-q\tquiet mode - do not print\n" \
 	"\t-P\tSet directory prefix to DIR\n" \
 	"\t-O\tsave to filename ('-' for stdout)\n" \
+	"\t-U\tadjust 'User-Agent' field\n" \
 	"\t-Y\tuse proxy ('on' or 'off')"
 
 #define which_trivial_usage \
diff -ur busybox-1.01/networking/wget.c busybox-1.01.new/networking/wget.c
--- busybox-1.01/networking/wget.c	2008-03-06 16:38:22.000000000 +1300
+++ busybox-1.01.new/networking/wget.c	2008-03-06 16:35:07.000000000 +1300
@@ -149,6 +149,7 @@
 #define WGET_OPT_HEADER	16
 #define WGET_OPT_PREFIX	32
 #define WGET_OPT_PROXY	64
+#define WGET_OPT_USER_AGENT    128
 
 static const struct option wget_long_options[] = {
 	{ "continue",        0, NULL, 'c' },
@@ -158,6 +159,7 @@
 	{ "header",	         1, NULL, 131 },
 	{ "directory-prefix",1, NULL, 'P' },
 	{ "proxy",           1, NULL, 'Y' },
+	{ "user-agent",      1, NULL, 'U' },
 	{ 0,                 0, 0, 0 }
 };
 
@@ -187,13 +189,14 @@
 	int quiet_flag = FALSE;		/* Be verry, verry quiet...			*/
 	int use_proxy = 1;          /* Use proxies if env vars are set  */
 	char *proxy_flag = "on";	/* Use proxies if env vars are set  */
+	char *user_agent = "Wget"; /* Content of the "User-Agent" header field */
 
 	/*
 	 * Crack command line.
 	 */
 	bb_opt_complementaly = "\203*";
 	bb_applet_long_options = wget_long_options;
-	opt = bb_getopt_ulflags(argc, argv, "cq\213O:\203:P:Y:", &fname_out, &headers_llist, &dir_prefix, &proxy_flag);
+	opt = bb_getopt_ulflags(argc, argv, "cq\213O:\203:P:Y:U:", &fname_out, &headers_llist, &dir_prefix, &proxy_flag, &user_agent);
 	if (opt & WGET_OPT_CONTINUE) {
 		++do_continue;
 	}
@@ -330,7 +333,7 @@
 				fprintf(sfp, "GET /%s HTTP/1.1\r\n", target.path);
 			}
 
-			fprintf(sfp, "Host: %s\r\nUser-Agent: Wget\r\n", target.host);
+			fprintf(sfp, "Host: %s\r\nUser-Agent: %s\r\n", target.host, user_agent);
 
 #ifdef CONFIG_FEATURE_WGET_AUTHENTICATION
 			if (target.user) {
