diff -Naur freetype-2.0.5.lnx/builds/unix/detect.mk freetype-2.0.5.cyg/builds/unix/detect.mk
--- freetype-2.0.5.lnx/builds/unix/detect.mk	Tue Jun 19 15:45:48 2001
+++ freetype-2.0.5.cyg/builds/unix/detect.mk	Fri Jan  4 07:11:44 2002
@@ -18,7 +18,8 @@
 
   # Note: this test is duplicated in "builds/toplevel.mk".
   #
-  is_unix := $(strip $(wildcard /sbin/init) $(wildcard /usr/sbin/init) $(wildcard /hurd/auth))
+#  is_unix := $(strip $(wildcard /sbin/init) $(wildcard /usr/sbin/init) $(wildcard /hurd/auth))
+  is_unix := unix
   ifneq ($(is_unix),)
 
     PLATFORM := unix
@@ -78,3 +79,4 @@
 endif   # test PLATFORM unix
 
 # EOF
+
