diff -Naur kaffe-1.0.6.org/config/config-std.h kaffe-1.0.6/config/config-std.h
--- kaffe-1.0.6.org/config/config-std.h	Thu Mar  9 21:53:11 2000
+++ kaffe-1.0.6/config/config-std.h	Sat Feb 16 17:47:42 2002
@@ -28,6 +28,8 @@
 #include <sys/resource.h>
 #endif
 #if defined(HAVE_WINNT_H)
+#include <windows.h>
+#include <windef.h>
 #include <winnt.h>
 #endif
 #if defined(HAVE_WINTYPES_H)
diff -Naur kaffe-1.0.6.org/kaffe/kaffevm/sha-1.h kaffe-1.0.6/kaffe/kaffevm/sha-1.h
--- kaffe-1.0.6.org/kaffe/kaffevm/sha-1.h	Thu Jul 22 02:19:11 1999
+++ kaffe-1.0.6/kaffe/kaffevm/sha-1.h	Sat Feb 16 19:36:08 2002
@@ -111,14 +111,14 @@
 /* I have taken some of this code from my MD5 implementation */
 
 #undef ROTATE
-#if defined(WIN32)
+#if defined(WIN32) && !defined(__GNUC__)
 #define ROTATE(a,n)     _lrotl(a,n)
 #else
 #define ROTATE(a,n)     (((a)<<(n))|(((a)&0xffffffff)>>(32-(n))))
 #endif
 
 /* A nice byte order reversal from Wei Dai <weidai@eskimo.com> */
-#if defined(WIN32)
+#if defined(WIN32) && !defined(__GNUC__)
 /* 5 instructions with rotate instruction, else 9 */
 #define Endian_Reverse32(a) \
 	{ \
