--- /ipkg_cmd.c.orig	2008-03-30 14:39:17.000000000 +0200
+++ /ipkg_cmd.c	2008-04-01 17:24:42.000000000 +0200
@@ -411,7 +411,6 @@
      pkg_vec_t *all;
      int i;
      pkg_t *pkg;
-     ipkg_intercept_t ic;
      int r, err = 0;
 
      ipkg_message(conf, IPKG_INFO,
@@ -421,8 +420,6 @@
      all = pkg_vec_alloc();
      pkg_hash_fetch_available(&conf->pkg_hash, all);
 
-     ic = ipkg_prep_intercepts (conf);
-    
      for(i = 0; i < all->len; i++) {
 	  pkg = all->pkgs[i];
 
@@ -445,10 +442,6 @@
 	  }
      }
 
-     r = ipkg_finalize_intercepts (ic);
-     if (r && !err)
-	 err = r;
-
      pkg_vec_free(all);
      return err;
 }
