--- ~/include/freetype/config/ftoption.h~	2008-09-01 16:37:39.000000000 +0200
+++ ~/include/freetype/config/ftoption.h	2008-09-01 16:37:39.000000000 +0200
@@ -375,7 +375,7 @@
   /* By undefining this, you will only compile the code necessary to load  */
   /* TrueType glyphs without hinting.                                      */
   /*                                                                       */
-#undef TT_CONFIG_OPTION_BYTECODE_INTERPRETER
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
