--- ~/examples/udhcp/simple.script~	2008-07-15 12:44:11.000000000 +0200
+++ ~/examples/udhcp/simple.script	2008-07-15 12:44:11.000000000 +0200
@@ -9,7 +9,9 @@
 
 case "$1" in
     deconfig)
-    /sbin/ifconfig $interface 0.0.0.0
+     # don't deconfig eth0 when running with nfsroot
+     cat /proc/cmdline | grep -q "root=/dev/nfs"
+     [ $? -ne 0 -o $interface != "eth0" ] && /sbin/ifconfig $interface 0.0.0.0                                           
     ;;
 
     renew|bound)
