--- /debianutils/start_stop_daemon.c.orig	2008-04-06 15:49:17.000000000 +0200
+++ /debianutils/start_stop_daemon.c	2008-04-06 19:15:52.000000000 +0200
@@ -51,15 +51,18 @@
 pid_is_exec(pid_t pid, const char *name)
 {
 	char buf[32];
-	struct stat sb, exec_stat;
-
-	if (name && stat(name, &exec_stat))
-		bb_perror_msg_and_die("stat %s", name);
+	FILE *f;
+	int c;
 
-	sprintf(buf, "/proc/%d/exe", pid);
-	if (stat(buf, &sb) != 0)
+	sprintf(buf, "/proc/%d/cmdline", pid);
+	f = fopen(buf, "r");
+	if (!f)
 		return 0;
-	return (sb.st_dev == exec_stat.st_dev && sb.st_ino == exec_stat.st_ino);
+
+	while ((c = getc(f)) != EOF && c && c == *name)
+		name++;
+	fclose(f);
+	return (*name == '\0');
 }
 
 static int
